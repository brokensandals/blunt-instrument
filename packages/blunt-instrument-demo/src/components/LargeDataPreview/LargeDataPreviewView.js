import React from 'react';
import './LargeDataPreview.css';

export default function LargeDataPreviewView({ data }) {
  switch (typeof data) {
    case 'boolean':
      return <p className="LargeDataPreview boolean">{data.toString()}</p>;
    case 'string':
      return <pre className="LargeDataPreview string">{data}</pre>;
    case 'number':
      return <p className="LargeDataPreview number">{data}</p>;
    default:
      return <pre className="LargeDataPreview object"><code>{JSON.stringify(data, null, 2)}</code></pre>;
  }
}
