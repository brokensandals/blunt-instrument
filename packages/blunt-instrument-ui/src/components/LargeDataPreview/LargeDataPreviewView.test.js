import React from 'react';
import renderer from 'react-test-renderer';
import LargeDataPreviewView from './LargeDataPreviewView';

describe('LargeDataPreviewView', () => {
  const cases = [
    ['boolean true', true],
    ['boolean false', false],
    ['short string', 'Hello World'],
    ['multiline string', "Hello, \"World\"!\nI am so pleased to exist and, secondarily, to make your acquaintance.\nBye now."],
    ['number', 1234567],
    ['object', { foo: 'bar' }],
  ];

  cases.forEach(([name, value]) => {
    test(name, () => {
      const result = renderer.create(<LargeDataPreviewView data={value} />).toJSON();
      expect(result).toMatchSnapshot();
    });
  });
});
