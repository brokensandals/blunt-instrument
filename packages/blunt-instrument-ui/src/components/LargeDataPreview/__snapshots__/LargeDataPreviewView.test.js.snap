// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LargeDataPreviewView boolean false 1`] = `
<p
  className="LargeDataPreview boolean"
>
  false
</p>
`;

exports[`LargeDataPreviewView boolean true 1`] = `
<p
  className="LargeDataPreview boolean"
>
  true
</p>
`;

exports[`LargeDataPreviewView multiline string 1`] = `
<pre
  className="LargeDataPreview string"
>
  Hello, "World"!
I am so pleased to exist and, secondarily, to make your acquaintance.
Bye now.
</pre>
`;

exports[`LargeDataPreviewView number 1`] = `
<p
  className="LargeDataPreview number"
>
  1234567
</p>
`;

exports[`LargeDataPreviewView object 1`] = `
<pre
  className="LargeDataPreview object"
>
  <code>
    {
  "foo": "bar"
}
  </code>
</pre>
`;

exports[`LargeDataPreviewView short string 1`] = `
<pre
  className="LargeDataPreview string"
>
  Hello World
</pre>
`;
